source/table_view.tsx

```diff
@@ -14,21 +16,51 @@ interface Properties {
 
   /** The CSS style to apply. */
   style?: any;
+
+  /** The width of active area measured from the right edge. */
+  activeWidth?: number;
 }
 
 /** Renders a TableModel to HTML. */
-export class TableView extends React.Component<Properties> {
+export class TableView extends React.Component<Properties> implements 
+    TableInterface {
   public static readonly defaultProps = {
     header: [] as string[],
-    style: {}
+    style: {},
+    activeWidth: 20
   };
 
+  constructor(props: Properties) {
+    super(props);
+    this.headerRefs = [];
+    for(let i = 0; i < this.props.labels.length; ++i) {
+      this.headerRefs[i] = null;
+    }
+  }
+
+  public componentDidMount() {
+    this.columnResizer = new ColumnResizer(this);
```

Comment:
Is there a reason why ```columnResizer``` is initialized here instead of the constructor?

Reply:
It's because when the state machine enters s0 it calls restoreCursor. The ref isn't initialized yet so if I initialize it in the constructor it crashes.